functions/Get-DbatoolsError.ps1

```diff
@@ -0,0 +1,67 @@
+function Get-DbatoolsError {
+    <#
+    .SYNOPSIS
+        Returns detailed dbatools-related error information
+
+    .DESCRIPTION
+        Returns detailed dbatools-related error information
+
+        By default, it only returns the most recent error ($error[0])
+
+    .PARAMETER First
+        Works like `Select-Object -First 1`
+
+    .PARAMETER Last
+        Works like `Select-Object -Last 1`
+
+    .PARAMETER Skip
+        Works like `Select-Object -Skip 1`
+
+    .PARAMETER All
+        Returns detailed information for all dbatools-related errors
+
+    .NOTES
+        Tags: Module, Support
+        Author: Chrissy LeMaire (@cl)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbatoolsError
+
+    .EXAMPLE
+        PS C:\> Get-DbatoolsError
+
+        Returns detailed error information for the most recent dbatools error
+
+    .EXAMPLE
+        PS C:\> Get-DbatoolsError -All
+
+        Returns detailed error information for all dbatools-related errors
+
+    .EXAMPLE
+        PS C:\> Get-DbatoolsError -Last 1
+
+        Returns the oldest dbatools-related error in the pipeline
+    #>
+    [CmdletBinding()]
+    param (
+        [int]$First,
+        [int]$Last,
+        [int]$Skip,
+        [switch]$All
+    )
+    process {
+        if (Test-Bound -Not First, Last, Skip, All) {
+            $First = 1
+        }
+
+        if ($All) {
+            $First = $global:error.Count
+        }
+
+        $global:error | Where-Object ScriptStackTrace -match dbatools | Select-Object -First $First -Last $Last -Skip $Skip -Property CategoryInfo, ErrorDetails, Exception, FullyQualifiedErrorId, InvocationInfo, PipelineIterationInfo, PSMessageDetails, ScriptStackTrace, TargetObject
```

Comment:
Why not call `Get-DbatoolsLog -Errors`

https://github.com/sqlcollaborative/dbatools/blob/development/functions/Get-DbatoolsLog.ps1

Reply:
Our errors are never detailed enough for me. They are pretty but not for deep diving stuff. Compare the screenshot above with a similar error using dbatoolslog 

![image](https://user-images.githubusercontent.com/8278033/126786961-0c5a2ae2-f27f-4fdd-a2b4-a92157b3886c.png)

Reply:
Well we can always change it. The log command retrieves errors based on `[Sqlcollaborative.Dbatools.Message.LogHost]::GetErrors()` which will be the errors for the primary session. If you query `$errors` for our stack name then you are also going to be pulling in the import errors and others that are always seen from code running async (in runspaces we generate). As long as users understand this when they use the new command.